MediaBrowser.Model/System/LibraryStorageInfo.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+
+namespace MediaBrowser.Model.System;
+
+/// <summary>
+/// Contains informations about a libraries storage informations.
+/// </summary>
+public class LibraryStorageInfo
+{
+    /// <summary>
+    /// Gets or sets the Library Id.
+    /// </summary>
+    public required Guid Id { get; set; }
+
+    /// <summary>
+    /// Gets or sets the name of the library.
+    /// </summary>
+    public required string Name { get; set; }
+
+    /// <summary>
+    /// Gets or sets the storage informations about the folders used in a library.
+    /// </summary>
+    public required IReadOnlyCollection<FolderStorageInfo> Folders { get; set; }
```

Comment:
Don't have a better place to comment this, can we also add the FolderStorageInfo collection to VirtualFolderInfo? (could be in a separate PR). It would probably deprecate the existing Locations array.

Reply:
i would really like to keep that seperate as much as possible. But yes if we decide to do it, that would be a seperate PR.